src/nvim/api/command.c

```diff
@@ -633,6 +633,24 @@ String nvim_cmd(uint64_t channel_id, Dict(cmd) *cmd, Dict(cmd_opts) *opts, Arena
   build_cmdline_str(&cmdline, &ea, &cmdinfo, args);
   ea.cmdlinep = &cmdline;
 
+  // set when Not Implemented
+  const int ni = is_cmd_ni(ea.cmdidx);
+
+  // Check for "++opt=val" argument.
+  if (ea.argt & EX_ARGOPT) {
+    while (ea.arg[0] == '+' && ea.arg[1] == '+') {
+      if (getargopt(&ea) == FAIL && !ni) {
+        api_set_error(err, kErrorTypeValidation, "Invalid argument");
+        goto end;
+      }
+    }
+  }
+
+  // Check for "+command" argument.
+  if ((ea.argt & EX_CMDARG) && !ea.usefilter) {
+    ea.do_ecmd_cmd = getargcmd(&ea.arg);
+  }
```

Comment:
Would it be possible to move this code from here and `do_one_cmd` to `execute_cmd0` so there isn't any code duplication?